alerting/src/test/kotlin/org/opensearch/alerting/resthandler/SecureMonitorRestApiIT.kt

```diff
@@ -870,14 +870,10 @@ class SecureMonitorRestApiIT : AlertingRestTestCase() {
 
     fun `test execute bucket-level monitor with user having partial index permissions`() {
 
-        createUserWithDocLevelSecurityTestDataAndCustomRole(
-            user,
-            TEST_HR_INDEX,
-            TEST_HR_ROLE,
-            TEST_HR_BACKEND_ROLE,
-            TERM_DLS_QUERY,
-            getClusterPermissionsFromCustomRole(ALERTING_FULL_ACCESS_ROLE)
-        )
+        createUser(user, user, arrayOf(TEST_HR_BACKEND_ROLE))
+        createTestIndex(TEST_HR_INDEX)
+        createCustomIndexRoleWithDocLevelSecurity(TEST_HR_ROLE, TEST_HR_INDEX, TERM_DLS_QUERY, getClusterPermissionsFromCustomRole(ALERTING_INDEX_MONITOR_ACCESS))
+        createUserRolesMapping(TEST_HR_ROLE, arrayOf(user))
```

Comment:
Do we also need tests to verify cross user accesses such as between UserA and UserB?

Reply:
different users were used to eliminate issues during parallel execution by the test engine while running tests, probably not required